pyqtorch/modules/circuit.py

```diff
@@ -85,8 +85,11 @@ def forward(self, state: torch.Tensor, thetas: torch.Tensor = None) -> torch.Ten
             torch.Tensor: The output quantum state tensor after applying the circuit operations.
 
         """
-        for op in self.operations:
+
+        def fn(op: torch.nn.Module) -> torch.Tensor:
             state = op(state, thetas)
+
+        map(fn, self.operations)
```

Comment:
`functools.reduce` would be the way to here I think, but won’t make a difference to the previous for loop I think?